tests/integration/aof-multi-part.tcl

```diff
@@ -0,0 +1,1065 @@
+            assert_equal 1 [check_file_exist $aof_dirpath "${aof_basename}.1${::incr_aof_sufix}${::aof_format_suffix}"]
+            assert_equal 1 [check_file_exist $aof_dirpath "${aof_basename}.2${::incr_aof_sufix}${::aof_format_suffix}"]
+
+            r config set aof-disable-auto-gc no
+
+            # Auto gc success
+            assert_aof_manifest_content $aof_manifest_filepath {
+                {file appendonly.aof.7.base.rdb seq 7 type b} 
+                {file appendonly.aof.3.incr.aof seq 3 type i}
+            }
+
+            # wait bio delete history 
+            wait_for_condition 1000 500 {
+                [check_file_exist $aof_dirpath "${aof_basename}.5${::base_aof_sufix}${::rdb_format_suffix}"] == 0 &&
+                [check_file_exist $aof_dirpath "${aof_basename}.6${::base_aof_sufix}${::rdb_format_suffix}"] == 0 &&
+                [check_file_exist $aof_dirpath "${aof_basename}.1${::incr_aof_sufix}${::aof_format_suffix}"] == 0 &&
+                [check_file_exist $aof_dirpath "${aof_basename}.2${::incr_aof_sufix}${::aof_format_suffix}"] == 0
+            } else {
+                fail "Failed to delete history AOF"
+            }
+        }
+
+        test "AOF reload can recovery the sequence" {
+            # Current manifest, BASE seq 7 and INCR seq 3
+            assert_aof_manifest_content $aof_manifest_filepath {
+                {file appendonly.aof.7.base.rdb seq 7 type b} 
+                {file appendonly.aof.3.incr.aof seq 3 type i}
+            }
+
+            r debug loadaof
+
+            # Trigger AOFRW
+            r bgrewriteaof
+            waitForBgrewriteaof r
+
+            # Now BASE seq is 8 and INCR seq is 4
+            assert_aof_manifest_content $aof_manifest_filepath {
+                {file appendonly.aof.8.base.rdb seq 8 type b} 
+                {file appendonly.aof.4.incr.aof seq 4 type i}
+            }
+        }
+
+        test "AOF enable during BGSAVE will not write data util AOFRW finish" {
+            r flushall
+            r config set rdb-key-save-delay 10000000
+            r config set appendonly no
+
+            r set k1 v1
+        
+            r bgsave
+            wait_for_condition 10 100 {
+                [s rdb_bgsave_in_progress] eq 1
+            } else {
+                fail "bgsave did not start in time"
+            }
+
+            # Make server.aof_rewrite_scheduled = 1
+            r config set appendonly yes
+
+            # Not open new INCR aof
+            assert_aof_manifest_content $aof_manifest_filepath {
+                {file appendonly.aof.8.base.rdb seq 8 type b} 
+                {file appendonly.aof.4.incr.aof seq 4 type i}
+            }
+
+            r set k2 v2
+            r debug loadaof
+
+            # Both k1 and k2 lost
+            assert_equal 0 [r exists k1]
+            assert_equal 0 [r exists k2]
+
+            r config set rdb-key-save-delay 0
+            waitForBgsave r
```

Comment:
1. let's make sure that all of the above steps got to complete while the `rdb_bgsave_in_progress` is still 1.
2. setting the key-save-delay will not cause the fork to detect it, and we'll still have to wait for it to complete. i guess we have to kill it to proceed.

Reply:
OK